app/controllers/match_controller.rb

```diff
@@ -5,6 +5,20 @@ def artworks
     render json: response
   end
 
+  def filter_artworks
+    valid_params = params.permit(
+      FilteredArtworkSearchService::ALLOWED_PARAMS,
+      # HACK: Per the Strong Parameters convention, array-valued params
+      # must be mapped to an empty array, so the allowlist above
+      # isn't quite sufficient. We therefore redundantly add…
+      gene_ids: [],
+      artist_ids: []
+    ).to_h
```

Comment:
The [Rails Strong Parameters](https://github.com/rails/strong_parameters) convention is in place by default in controllers.

But it doesn't play nice with the array-valued params that we [pass with old-style Rack encoding](https://github.com/artsy/rosalind/blob/412e361c9feb2dbb6b93783c0a9605d50a2f4274/app/services/filtered_artwork_search_service.rb#L6) (a Gravity constraint?).

So this hack redundantly permits the square bracket version of these array params ( `gene_ids[]` in addition to `gene_ids`)

Alternatively, we could avoid this little dance by boldly permitting all parameters with…

```ruby
params.permit!
valid_params = params.to_h.without('controller', 'action')
```

…which is less ugly, and safe enough. There's no mass assignment happening here, which is what Strong Params is meant to protect.

I'm fine with either of these imperfect approaches, but if there's a better one I'd love to know!

Reply:
I like the approach you have here - I'm always in favor of being a little more paranoid about params, and I don't think this is the type of hack that's likely to bite us in the future (especially with the nice comment you wrote to clarify what's happening).